docs/design.md

```diff
@@ -0,0 +1,3 @@
+## Top level design
```

Comment:
Do we really need to split the .md into multiple files. I thought we can get specific lines from the same .md file

Reply:
I broke up the sections into their own pages since the table of contents on the left side-bar uses references to separate pages and because this way the table of contents does not have to be duplicated in the index where the main content is.

Here is a preview: https://advanced-micro-devices-rpp--145.com.readthedocs.build/en/145/index.html